src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -54,6 +56,23 @@ public FileMonitoringService(IOptions<ScriptJobHostOptions> scriptOptions, ILogg
 
             _shutdown = Shutdown;
             _shutdown = _shutdown.Debounce(milliseconds: 500);
+            _rootDirectorySnapshot = GetDirectorySnapshot();
+        }
+
+        internal ImmutableArray<string> GetDirectorySnapshot()
+        {
+            if (_scriptOptions.RootScriptPath != null)
+            {
+                try
+                {
+                    return Directory.EnumerateDirectories(_scriptOptions.RootScriptPath).ToImmutableArray();
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    return ImmutableArray<string>.Empty;
```

Comment:
This can be removed... just catch (log perhaps) and let the execution flow. No need for the duplication here